generator/collector_mini_roundabout.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "generator/collector_interface.hpp"
+#include "generator/mini_roundabout_info.hpp"
+
+#include "coding/file_writer.hpp"
+
+#include <cstdint>
+#include <fstream>
+#include <functional>
+#include <memory>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace generator
+{
+namespace cache
+{
+class IntermediateDataReader;
+}  // namespace cache
+}  // namespace generator
+
+struct OsmElement;
+
+namespace feature
+{
+class FeatureBuilder;
+}  // namespace feature
+
+namespace routing
+{
+class MiniRoundaboutProcessor
+{
+public:
+  using Fn = std::function<void(MiniRoundaboutInfo const &)>;
+
+  MiniRoundaboutProcessor(std::string const & filename);
+  ~MiniRoundaboutProcessor();
+
+  void ForEachMiniRoundabout(Fn && toDo) const;
+  void ProcessNode(OsmElement const & element);
+  void ProcessWay(OsmElement const & element);
+  void ProcessRestriction(uint64_t osmId);
+
+  void FillMiniRoundaboutsInWays();
+
+  void Finish();
+  void Merge(MiniRoundaboutProcessor const & MiniRoundaboutProcessor);
+
+private:
+  std::string m_waysFilename;
+  std::unique_ptr<FileWriter> m_waysWriter;
+  std::unordered_map<uint64_t, MiniRoundaboutInfo> m_miniRoundabouts;
+  std::set<uint64_t> m_miniRoundaboutsExceptions;
+};
+
+class MiniRoundaboutCollector : public generator::CollectorInterface
+{
+public:
+  explicit MiniRoundaboutCollector(std::string const & filename);
+
```

Comment:
`// CollectorInterface overrides:`?

Reply:
https://github.com/mapsme/omim/pull/11863#discussion_r333000370

Reply:
я писал про коммент 
```
// We will process all nodes before ways because of o5m format:
```

обычно коммент ревьюера относиться к последней строке в снепшоте с кодом на гитхабе

Reply:
А, поняла. Исправила.